src/cljs/athens/views/athena.cljs

```diff
@@ -272,76 +229,121 @@
-                                                                                               (navigate-uid uid)))
-                                                                                 :class    (when (= i index) "selected")})
-
-                                                   [:div (use-style result-body-style)
-                                                    [:h4.title (use-sub-style result-style :title)
-                                                     [:b "Create Page: "]
-                                                     query]]
-                                                   [:span.link-leader (use-sub-style result-style :link-leader) [(r/adapt-react-class Create)]]]
-
-                                                  [:div (use-style result-style {:key      i
-                                                                                 :on-click (fn []
-                                                                                             (let [selected-page {:node/title   title
-                                                                                                                  :block/uid    uid
-                                                                                                                  :block/string string
-                                                                                                                  :query        query}]
-                                                                                               (dispatch [:athena/toggle])
-                                                                                               (dispatch [:athena/update-recent-items selected-page])
-                                                                                               (navigate-uid uid)))
-                                                                                 :class    (when (= i index) "selected")})
-                                                   [:div (use-style result-body-style)
-
-                                                    [:h4.title (use-sub-style result-style :title) (highlight-match query title)]
-                                                    (when string
-                                                      [:span.preview (use-sub-style result-style :preview) (highlight-match query string)])]
-                                                   [:span.link-leader (use-sub-style result-style :link-leader) [(r/adapt-react-class ArrowForward)]]])))]))]])))})))
+  (let [s (r/atom {:index   0
+                   :query   nil
+                   :results []})
+        search-handler (create-search-handler s)]
+    (fn []
+      (let [open? @(subscribe [:athena/open])]
+        (when open?
+          [view-utils/track-outside-click #(dispatch [:athena/toggle])
+           [athena-keybindings s
```

Comment:
Why keybindings get involved with state here?

Reply:
They need access to change the currently selected element in Athena. That is part of the current implementation